core/src/test/scala/unit/kafka/network/SocketServerTest.scala

```diff
@@ -274,8 +274,8 @@ class SocketServerTest {
     sendRequest(plainSocket, apiVersionRequestBytes(clientId, version))
     var receivedReq = receiveRequest(server.dataPlaneRequestChannel)
 
-    assertEquals(ClientInformation.UNKNOWN_NAME_OR_VERSION, receivedReq.context.clientInformation.softwareName)
-    assertEquals(ClientInformation.UNKNOWN_NAME_OR_VERSION, receivedReq.context.clientInformation.softwareVersion)
+    assertEquals(expectedClientSoftwareName, receivedReq.context.clientInformation.softwareName)
+    assertEquals(expectedClientSoftwareVersion, receivedReq.context.clientInformation.softwareVersion)
```

Comment:
The comment above these lines is wrong with this change. I assume there was a reason for the original behavior though - @dajac @rajinisivaram do you recall?